drivers/gpio/gpio_pcal64xxa.c

```diff
@@ -0,0 +1,955 @@
+		return -EIO;
+	}
+
+	rc = pcal64xxa_i2c_write(i2c, PCAL6416A_REG_INPUT_LATCH_1, (uint8_t)(input_latch >> 8));
+	if (rc != 0) {
+		LOG_ERR("failed to configure input latch: %d", rc);
+		return -EIO;
+	}
+
+	rc = pcal64xxa_i2c_write(i2c, PCAL6416A_REG_INTERRUPT_MASK_0, (uint8_t)interrupt_mask);
+	if (rc != 0) {
+		LOG_ERR("failed to configure interrupt mask: %d", rc);
+		return -EIO;
+	}
+
+	rc = pcal64xxa_i2c_write(i2c, PCAL6416A_REG_INTERRUPT_MASK_1,
+				 (uint8_t)(interrupt_mask >> 8));
+	if (rc != 0) {
+		LOG_ERR("failed to configure interrupt mask: %d", rc);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int pcal6416a_reset_state_apply(const struct i2c_dt_spec *i2c)
+{
+	int rc;
+	static const uint8_t reset_state[][2] = {
+		{PCAL6416A_REG_POLARITY_INVERSION_0, 0},
+		{PCAL6416A_REG_POLARITY_INVERSION_1, 0},
+		{PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_0_0, 0xff},
+		{PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_0_1, 0xff},
+		{PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_1_0, 0xff},
+		{PCAL6416A_REG_OUTPUT_DRIVE_STRENGTH_1_1, 0xff},
+		{PCAL6416A_REG_OUTPUT_PORT_CONFIGURATION, 0},
+	};
+
+	for (int i = 0; i < ARRAY_SIZE(reset_state); ++i) {
+		rc = pcal64xxa_i2c_write(i2c, reset_state[i][0], reset_state[i][1]);
+		if (rc != 0) {
+			LOG_ERR("failed to reset register %02x: %d", reset_state[i][0], rc);
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static const struct pcal64xxa_chip_api pcal6416a_chip_api = {
+	.pins_cfg_apply = pcal6416a_pins_cfg_apply,
+	.triggers_apply = pcal6416a_triggers_apply,
+	.inputs_read = pcal6416a_inputs_read,
+	.outputs_write = pcal6416a_outputs_write,
+	.reset_state_apply = pcal6416a_reset_state_apply,
+};
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(nxp_pcal6416a) */
+
+int pcal64xxa_init(const struct device *dev)
+{
+	const struct pcal64xxa_drv_cfg *drv_cfg = dev->config;
+	struct pcal64xxa_drv_data *drv_data = dev->data;
+	const struct pcal64xxa_pins_cfg initial_pins_cfg = {
+		.configured_as_inputs = 0xFFFF,
+		.outputs_high = 0,
+		.pull_ups_selected = 0,
+		.pulls_enabled = 0,
+	};
+	const struct pcal64xxa_triggers initial_triggers = {
+		.masked = 0xFFFF,
+	};
+	int rc;
+	uint16_t int_sources;
+
+	LOG_DBG("initializing PCAL64XXA");
+
+	if (drv_cfg->ngpios != 8U && drv_cfg->ngpios != 16U) {
+		LOG_ERR("Invalid value ngpios=%u. Expected 8 or 16!", drv_cfg->ngpios);
+		return -EINVAL;
+	}
+
+	/*
+	 * executing the is ready check on i2c_bus_dev instead of on i2c.bus
+	 * to avoid a const warning
+	 */
+	if (!device_is_ready(drv_cfg->i2c.bus)) {
+		LOG_ERR("%s is not ready", drv_cfg->i2c.bus->name);
+		return -ENODEV;
+	}
+
+	/* If the RESET line is available, use it to reset the expander.
+	 * Otherwise, write reset values to registers that are not used by
+	 * this driver.
+	 */
+	if (drv_cfg->reset_gpio_dev) {
+		if (!device_is_ready(drv_cfg->reset_gpio_dev)) {
+			LOG_ERR("%s is not ready", drv_cfg->reset_gpio_dev->name);
+			return -ENODEV;
+		}
+
+		rc = gpio_pin_configure(drv_cfg->reset_gpio_dev, drv_cfg->reset_gpio_pin,
```

Comment:
Please use gpio DT specs and switch to `gpio_pin_configure_dt()` and friends.

Reply:
done